encode.go

```diff
@@ -58,12 +58,15 @@ func (e *MarshalInvalidTypeError) Error() string {
 // stream.
 type Encoder struct {
 	w *bufio.Writer
+
+	LineEnd []byte
```

Comment:
I think I would prefer `LineEnd` to be un-exported and a `SetLineEnd` method added.